eng/build.yml

```diff
@@ -84,7 +84,6 @@ jobs:
     - _PhaseName : ${{ parameters.name }}
     - _HelixType: build/product
     - _HelixBuildConfig: $(_BuildConfig)
-    - _Pipeline_StreamDumpDir: $(Build.SourcesDirectory)/artifacts/tmp/$(_BuildConfig)/streams
```

Comment:
It looked to me like just a yaml variable used in the publish artifacts line and I like the path to be in the step.  It is harder to "read" for it to be in a var at the top of the file.

Reply:
Yaml variables get injected as environment variables in AzDO and it is where the sreamdumper will dump the raw stream in case of failure: https://github.com/dotnet/diagnostics/blob/69849b67ef0f2d07c29bfb73ade6c40c163e8a60/src/tests/eventpipe/common/StreamProxy.cs#L100

We need that if we want to recover them to reproduce the failure locally in case CI fails.